graphs/growth.py

```diff
@@ -36,55 +35,82 @@ def TG_formula(TG, A):
-
-with open(args.data, 'r') as stream:
-    try:
-        data = yaml.safe_load(stream)
-        active = [point['positive'] - point['recovered'] - point['dead'] for point in data]
-        dead = [point['dead'] for point in data]
-        recovered = [point['recovered'] for point in data]
-        date_list = [point['date'] for point in data]
-    except yaml.YAMLError as exc:
-        raise exc
-
-cumulative_active = list(filter(lambda x: x >= country.case_count, itertools.accumulate(active)))
-date_list = date_list[len(active) - len(cumulative_active):]
-x = np.arange(1, len(cumulative_active) * 2)
-y = country.formula.lambd(x)
-
-last_date = datetime.strptime(date_list[-1], '%Y-%m-%d')
-date_list += [(last_date + timedelta(days=d)).strftime('%Y-%m-%d') for d in range(1, 51)]
-
-layout = Layout(title=f"Active cases in {country.name}",
-                xaxis=dict(autorange=True,
-                           title=r'$\text{Days since the 200}^\mathrm{th}\text{ case}$'),
-                yaxis=dict(type='log', autorange=True, title='COVID-19 active cases'),
-                hovermode='x',
-                font={'size': 15},
-                legend=dict(x=0.01, y=0.99, borderwidth=1))
-
-figure = Figure(layout=layout)
-figure.add_trace(
-    Scatter(x=x,
-            y=cumulative_active,
-            mode='lines+markers',
-            name=f"Active cases",
-            line={'width': 3},
-            marker={'size': 8}))
-
-figure.add_trace(
-    Scatter(
-        x=x,
-        y=y,
-        text=date_list,
-        mode='lines',
-        name=country.formula.text,
-        line={
-            'dash': 'dash',
-            'width': 3
-        },
-    ))
-
-# offline.plot(figure, filename='graph.html')
-figure.show()
+
+class CountryData:
+    def __init__(self, path, name, country_basic):
+        self.name = name
+        self.country_basic = country_basic
+        self.path = os.path.join(path, 'data-' + name + '.yaml')
+        with open(self.path, 'r') as stream:
+            try:
+                data = yaml.safe_load(stream)
+                self.active = [point['positive'] - point['recovered'] - point['dead'] for point in data]
+                self.dead = [point['dead'] for point in data]
+                self.recovered = [point['recovered'] for point in data]
+                self.date_list = [point['date'] for point in data]
+            except yaml.YAMLError as exc:
+                raise exc
+
+        self.cumulative_active = list(filter(lambda x: x >= self.country_basic.case_count, itertools.accumulate(self.active)))
+        self.date_list = self.date_list[len(self.active) - len(self.cumulative_active):]
+        self.x = np.arange(1, len(self.cumulative_active) * 2)
+        self.y = self.country_basic.formula.lambd(self.x)
+
+        self.last_date = datetime.strptime(self.date_list[-1], '%Y-%m-%d')
+        self.date_list += [(self.last_date + timedelta(days=d)).strftime('%Y-%m-%d') for d in range(1, 51)]
```

Comment:
Tu je magická konštanta 50, ale od tej som už vlastne upustil a berie sa 2-krát dlhšie pole x ako je veľkosť dát (to je na to, aby bolo dobre vidno trend). Malo by sa to zjednotiť, čiže dĺžka x musí byť taká istá ako self.date_list.

Reply:
Done